src/test/scala/mesosphere/mesos/ResourceMatcherTest.scala

```diff
@@ -1179,6 +1179,60 @@ class ResourceMatcherTest extends UnitTest with Inside {
       resourceMatchResponse shouldBe a[ResourceMatchResponse.NoMatch]
       resourceMatchResponse.asInstanceOf[ResourceMatchResponse.NoMatch].reasons.head shouldEqual DeclinedScarceResources
     }
+
+    "support application-specific override of gpu scheduling behavior from restricted -> unrestricted" in {
+      val gpuConfig = AllConf.withTestConfig(
+        "--draining_seconds", "300",
+        "--gpu_scheduling_behavior", "restricted",
+        "--enable_features", "gpu_resources")
+      val offer = MarathonTestHelper.makeBasicOffer(gpus = 4)
+        .build()
+      val app = AppDefinition(
+        id = "/test".toRootPath,
+        resources = Resources(cpus = 1.0, mem = 128.0, disk = 0.0, gpus = 0),
+        portDefinitions = PortDefinitions(0, 0),
+        labels = Map("GPU_SCHEDULING_BEHAVIOR" -> "unrestricted")
+      )
+
+      val resourceMatchResponse = ResourceMatcher.matchResources(
+        offer,
+        app,
+        knownInstances = Seq.empty,
+        unreservedResourceSelector,
+        gpuConfig,
+        Seq.empty
+      )
+
+      resourceMatchResponse shouldBe a[ResourceMatchResponse.Match]
+    }
+
+    "support application-specific override of gpu scheduling behavior from unrestricted -> restricted" in {
+      val gpuConfig = AllConf.withTestConfig(
+        "--draining_seconds", "300",
+        "--gpu_scheduling_behavior", "unrestricted",
+        "--enable_features", "gpu_resources")
+      val offer = MarathonTestHelper.makeBasicOffer(gpus = 4)
+        .build()
+      val app = AppDefinition(
+        id = "/test".toRootPath,
+        resources = Resources(cpus = 1.0, mem = 128.0, disk = 0.0, gpus = 0),
+        portDefinitions = PortDefinitions(0, 0),
+        labels = Map("GPU_SCHEDULING_BEHAVIOR" -> "restricted")
```

Comment:
I think we are missing tests for no label. What do you think about using a property table?

```scala
Table(
    ("gpu_scheduling_behavior", "GPU_SCHEDULING_BEHAVIOR", expected),
    ("unrestricted", Some("restricted"), Match),
    ("unrestricted", None, Match),
    ("restricted", Some("unrestricted"), NoMatch),
    ("restricted", None, NoMatch),
)
```

This seems easier to comprehend to me.